### PR TITLE
Check Cached Remote Packages optimizations

### LINKED ISSUE #1558: Improve `npm view` command performance for remote repositories
# Task Description

The `npm install @angular/core` (for example) execution currently can take about 20 seconds. The problem related with `NpmRepositoryFeatures.countPackages` method, SQL query that happens there counts `ArtifactEntries`  to check if there are cached entities or not. We can rework this by using something like `SELECT 1 FROM ArtifactEntry WHERE ... LIMIT 1`. Also we should be sure that new query triggers `RepositoryArtifactIdGroupEntry` index.

# Tasks

The following tasks will need to be carried out:
* [ ] Create `NpmRepositoryFeatures.isThereAnyAlreadyCachedRemotePackages` method (maybe with better name :) )
* [ ] Drop `NpmRepositoryFeatures.countPackages` method

# Help

* [Our chat](https://chat.carlspring.org/)
* Points of contact:
  * @carlspring
  * @fuss86
  * @sbespalov
  * @steve-todorov

### PR DESCRIPTION
**Pull request by anki2189**
# Pull Request Description

This pull request closes #1558 

# Acceptance Test

* [ ] Building the code with `mvn clean install -Dintegration.tests` still works.
* [x] Running `mvn spring-boot:run` in the `strongbox-web-core` still starts up the application correctly.
* [x] Building the code and running the `strongbox-distribution` from a `zip` or `tar.gz` still works.
* [ ] The tests in the [`strongbox-web-integration-tests`](https://github.com/strongbox/strongbox-web-integration-tests/) still run properly.

# Questions

* Does this pull request break backward compatibility? 
  * [ ] Yes
  * [x] No

* Does this pull request require other pull requests to be merged first? 
  * [ ] Yes, please see #...
  * [x] No

* Does this require an update of the documentation?
  * [ ] Yes, please see strongbox/strongbox-docs#{PR_NUMBER}
  * [x] No

### CODE REVIEW

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-npm-layout-provider/src/main/java/org/carlspring/strongbox/repository/NpmRepositoryFeatures.java [hunk 1]
```diff
@@ -500,21 +501,23 @@ public void handle(RemoteRepositorySearchEvent event)
 
     }
 
-    private Long countPackages(String storageId,
-                               String repositoryId,
-                               Predicate predicate)
+    public Boolean anyRemotePackageCached(String storageId,
+                                                 String repositoryId)
```
Comment (carlspring) [inline strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-npm-layout-provider/src/main/java/org/carlspring/strongbox/repository/NpmRepositoryFeatures.java]: Cosmetic: Please, re-align the method parameters.

Comment (anki2189) [reply strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-npm-layout-provider/src/main/java/org/carlspring/strongbox/repository/NpmRepositoryFeatures.java]: Fixed

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-npm-layout-provider/src/main/java/org/carlspring/strongbox/repository/NpmRepositoryFeatures.java [hunk 2]
```diff
@@ -478,15 +480,14 @@ public void handle(RemoteRepositorySearchEvent event)
                 return;
             }
 
-            Predicate predicate = event.getPredicate();
```
Comment (sbespalov) [inline strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-npm-layout-provider/src/main/java/org/carlspring/strongbox/repository/NpmRepositoryFeatures.java]: @anki2189 you shouldn't drop this `predicate` here. Anyway we should not proceed with this issue (see [comment]())

**Comment by sbespalov** [comment]
I've added some details in #1558

**Comment by sbespalov** [comment]
@anki2189 thanks for looking into this, we can back to it some time later.
For now I'm closing this PR.

**Pull request closed without merging**